src/registry/GithubRegistry.ts

```diff
@@ -249,7 +252,12 @@ export class GithubRegistry extends BaseRegistry implements IRegistry {
 
   protected async fetch(url: string): Promise<Response> {
     this.logger.debug(`Fetching from github: ${url}`);
-    const response = await fetch(url);
+    const useToken = !(this.proxyUrl && url.startsWith(this.proxyUrl)) && !!this.authToken;
```

Comment:
Optional Nit: This is a quite hard to reason about condition, consider renaming some of the subconditions, something like `!proxied & hasToken`